### DISCUSSION TITLE
Convert PDF (CMYK) to CMYK image directly without loss of color

### DISCUSSION THREAD

**Discussion by SomenteRafael** [dlemstra/Magick.NET#704] [Help] [answered]
I'm tring convert a PDF (CMYK) to CMYK TIFF but seems like it is converted to RGB and back to CMYK. I'm using a PDF with an horizontal gradient from black to white and the result is a yellowish image. When I convert to PNG (RGB) instead of TIFF or JPG the image color is better, but it's still different, with less color nuance.

        public static void PDFTo(string input, string output)
        {
            var settings = new MagickReadSettings
            {
                Density = new Density(300, 300),
                ColorSpace = ColorSpace.CMYK
            };
            settings.ColorType = ColorType.ColorSeparation;
            using var images = new MagickImageCollection();
            images.Read(input, settings);

            var page = 1;
            foreach (var image in images)
            {
                image.ColorSpace = ColorSpace.CMYK;
                image.Format = MagickFormat.Tiff; // MagickFormat.jpg
                image.Write(output + input[input.LastIndexOf('\\')..^4] + "_" + page + ".tiff"); // ".jpg"
                page++;
            }
        }

How can I fix it? Is it possible? Is it a bug? Its viewers' bug and its all right with the images?

<table>
<tr>
<th>Input file</th><th><a href="https://github.com/dlemstra/Magick.NET/files/4955028/Rafael_100_100.pdf">PDF</a>
</th>
</tr>
<tr>
<th>PNG result</th><th><img width="100px" src="https://user-images.githubusercontent.com/43616612/88084288-8f4e7400-cb5a-11ea-81de-370223b4a5c7.png">
</th>
</tr>
<tr>
<th>JPG result</th><th><img width="100px" src="https://user-images.githubusercontent.com/43616612/88084301-95dceb80-cb5a-11ea-9f83-9a6821fef9eb.jpg">
</th>
</tr>
<tr>
<th>Colorfull input for tests</th><th><a href="https://github.com/dlemstra/Magick.NET/files/4965415/SOSample.pdf">PDF</a></th>
</tr>
</table>

Maybe its not wrong, I don't know.

**Comment by dlemstra** [comment, marked as answer]
It took some time to figure out why this is happening. The problem here is that the `PDF` decoder of ImageMagick used an option called `-dUseCIEColor` that would give you this strange output. After some reading we found that we no longer need this option and that is why we removed it. This patch will become available in the next release and you will then get the following output:

![test](https://user-images.githubusercontent.com/10426229/88460204-e75ae280-ce9a-11ea-823e-b491dbe09ca1.png)

I will try to publish that release this weekend.
